spec/cohort.spec.js

```diff
@@ -0,0 +1,472 @@
+const Cohorts = require('../src/cohort.js')
+describe('CoHORTS', () => {
+  let cohortLi
+
+  beforeEach(() => {
+    cohortLi = new Cohorts()
+  })
+  describe('addCohort', () => {
+    it('Add new cohort', () => {
+      const expected = [{ 'Cohort 1': [] }]
+
+      const result = cohortLi.addCohort('Cohort 1')
+
+      expect(result).toEqual(expected)
+    })
+    it('Trying to create new Cohort, but name already exist', () => {
+      const expected = 'Error: The name already exists'
+      cohortLi.addCohort('Cohort 1')
+      expect(() => cohortLi.addCohort('Cohort 1')).toThrowError(expected)
+    })
+    it('Name is empty string', () => {
+      const expected = 'Error: Please provide a Cohort name'
+      expect(() => cohortLi.addCohort('')).toThrowError(expected)
+    })
+    it('Cohort list is empty', () => {
+      const expected = []
+      const result = cohortLi.cohortList
+      expect(result).toEqual(expected)
+    })
+  })
+  describe('searchByName', () => {
+    it('Search cohort by name', () => {
+      const expected = { 'Cohort 1': [] }
```

Comment:
It this your structure of your local datastore?

```
this.cohortList = []
```

I also have a similar structure where I have an array as a value to an object key, and fill it with students belonging to a cohort.